services/blockchain-indexer/shared/dataService/business/transactionsEstimateFees.js

```diff
@@ -201,18 +203,41 @@ const estimateTransactionFees = async params => {
 	const transactionFeeEstimates = {
 		minFee: Number(minFee) + (BUFFER_BYTES_LENGTH * feeEstimatePerByte.minFeePerByte),
 		accountInitializationFee: await calcAccountInitializationFees(formattedTransaction),
-		messageFee: await calcMessageFee(formattedTransaction),
 	};
 
-	const dynamicFeeEstimates = calcDynamicFeeEstimates(
-		feeEstimatePerByte,
-		transactionFeeEstimates.minFee,
-		size,
-	);
-
 	estimateTransactionFeesRes.data = {
-		transactionFeeEstimates,
-		dynamicFeeEstimates,
+		transaction: {
+			fee: {
+				tokenID: feeEstimatePerByte.feeTokenID,
+				minimum: transactionFeeEstimates.minFee,
+			},
+			params: {
+				messageFee: await calcMessageFee(formattedTransaction),
+			},
+		},
+	};
+
+	// Priority available only when fee values are not 0
+	if (feeEstimatePerByte.low !== 0 || feeEstimatePerByte.med !== 0
+		|| feeEstimatePerByte.high !== 0) {
+		const dynamicFeeEstimates = calcDynamicFeeEstimates(
+			feeEstimatePerByte, transactionFeeEstimates.minFee, size);
+
+		estimateTransactionFeesRes.transaction.fee.priority = dynamicFeeEstimates;
+	}
+
+	const posConstants = await getPosConstants();
+
+	estimateTransactionFeesRes.meta = {
+		feeBreakdown: {
+			minimum: {
+				byteFee: feeEstimatePerByte.minFeePerByte,
+				additionalFees: {
+					registrationFee: posConstants.data.validatorRegistrationFee,
+					accountInitializationFee: transactionFeeEstimates.accountInitializationFee.amount,
+				},
+			},
+		},
 	};
```

Comment:
a bit off-topic: would it be better encapsulation to put fee related logics in fee-estimator microservice and expose an endpoint to get calculated values?

Reply:
Since fee calculation is only used for `estimate fees endpoint` and will not be used in any other microservice or endpoint, I feel better to leave it as it is since moving it to fee estimator microservice will introduce unnecessary latency.

Reply:
Let's not pollute the fee microservice. The fees microservice is solely responsible to run the dynamic fee algorithm to analyze the network traffic and determine the priority fees.